datasets/off_category/off_category.py

```diff
@@ -0,0 +1,80 @@
+from collections import namedtuple
+from dataclasses import dataclass
+import gzip
+import orjson
+from typing import Any, Dict
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+
+@dataclass
+class Feature:
+  spec: tfds.features.FeatureConnector
+  default_value: Any
+  input_field: str = None  # use feature name if None
+
+
+_DESCRIPTION = """
+Open Food Facts product category classification dataset.
+"""
+
+_RELEASE_NOTES = {
+  '0.0.1': 'Initial test'
+}
+
+# Don't forget to run `tfds build --register_checksums` when changing the data source
+_DATA_URL = 'https://github.com/openfoodfacts/openfoodfacts-ai/releases/download/dataset-category-2021-09-15/category_xx.{split}.jsonl.gz'
+
+_FEATURES = {
+  'code': Feature(tfds.features.Tensor(shape=(), dtype=tf.string), ''),
+  'product_name': Feature(tfds.features.Tensor(shape=(), dtype=tf.string), ''),
+  'ingredients_tags': Feature(tfds.features.Tensor(shape=(None,), dtype=tf.string), [],
```

Comment:
I'm chose to decorrelate data fields and feature names, because:
- Some future features won't exactly match data fields anyway (e.g. for nutriments we'll extract some sub-fields as independent features, and probably have a catch-all feature with the rest of the sub-fields, like we discussed with @alexgarel a few meetings ago).
- Model Inputs are based on the feature name, so we'd have to update the Inputs names when changing target fields. For instance, when updating to the new dataset, we'll go from 'known_ingredients_tags' to 'ingredients_tags'. If we decorrelate field/feature names like here we won't have to do anything model-side. Otherwise we'll have to update the model. Maybe it's better since it's explicit, or it's just more cumbersome.

Any thoughts about this?

Reply:
This PR si a great addition to improve scalability of the model!
I think decorrelating field / feature will enhance the reusability of the model ;)